flink-ml-iteration/src/main/java/org/apache/flink/iteration/datacache/nonkeyed/Segment.java

```diff
@@ -18,38 +18,80 @@
 
 package org.apache.flink.iteration.datacache.nonkeyed;
 
+import org.apache.flink.annotation.Internal;
 import org.apache.flink.core.fs.Path;
+import org.apache.flink.core.memory.MemorySegment;
 
-import java.io.Serializable;
+import java.util.List;
 import java.util.Objects;
 
-/** A segment represents a single file for the cache. */
-public class Segment implements Serializable {
+import static org.apache.flink.util.Preconditions.checkArgument;
+import static org.apache.flink.util.Preconditions.checkNotNull;
+import static org.apache.flink.util.Preconditions.checkState;
 
+/** A segment contains the information about a cache unit. */
+@Internal
+public class Segment {
+
+    /** The path to the file containing cached records. */
     private final Path path;
 
-    /** The count of the records in the file. */
+    /** The count of the records in the segment. */
```

Comment:
By `in the segment`, do you mean `in the file` or `in the memory`? Could you explain this in the Java doc?